File: fornituras-api/src/main/java/com/numobiz/solutions/fornituras/common/text/CodeNormalizer.java
package com.numobiz.solutions.fornituras.common.text;

import java.util.Locale;

/**
 * Normaliza códigos opacos (QR/serie) para comparar unicidad física de forma robusta: recorta,
 * elimina espacios y guiones internos y pasa a mayúsculas. Así "FOR-00001", "for 00001" y
 * "FOR00001" se consideran el mismo identificador y no generan duplicados "aparentemente
 * distintos".
 *
 * <p>A diferencia de {@link NameNormalizer} (nombres legibles: minúsculas, sin acentos), aquí los
 * códigos van en mayúsculas y se descartan separadores, porque el grabado/impresión puede variar.
 */
public final class CodeNormalizer {

	private CodeNormalizer() {
	}

	public static String normalize(String raw) {
		if (raw == null) {
			return "";
		}
		return raw.replaceAll("[\\s-]+", "").toUpperCase(Locale.ROOT);
	}
}

File: fornituras-api-dotnet/src/Fornituras.Api/Common/Text/CodeNormalizer.cs
namespace Fornituras.Api.Common.Text;

/// <summary>
/// Normaliza códigos opacos (QR/serie): recorta, elimina espacios/guiones y pasa a mayúsculas.
/// </summary>
public static class CodeNormalizer
{
    public static string Normalize(string? raw)
    {
        if (raw is null)
        {
            return string.Empty;
        }

        return System.Text.RegularExpressions.Regex.Replace(raw, @"[\s-]+", string.Empty)
            .ToUpperInvariant();
    }
}
